lorien/UI/Dialogs/SettingsDialog.gd

```diff
@@ -52,17 +47,34 @@ func _set_values() -> void:
 		Types.AAMode.NONE: _aa_mode.selected = AA_NONE_INDEX
 		Types.AAMode.OPENGL_HINT: _aa_mode.selected = AA_OPENGL_HINT_INDEX
 		Types.AAMode.TEXTURE_FILL: _aa_mode.selected = AA_TEXTURE_FILL_INDEX
-	match language:
-		Types.LOCALE_ENGLISH: _language_options.selected = LANGUAGE_ID_ENGLISH
-		Types.LOCALE_GERMAN: _language_options.selected = LANGUAGE_ID_GERMAN
-		Types.LOCALE_ITALIAN: _language_options.selected = LANGUAGE_ID_ITALIAN
-		Types.LOCALE_SPANISH: _language_options.selected = LANGUAGE_ID_SPANISH
+	
+	_set_languages(locale)
 	
 	_brush_size.value = brush_size
 	_brush_color.color = brush_color
 	_canvas_color.color = canvas_color
 	_project_dir.text = project_dir
 
+# -------------------------------------------------------------------------------------------------
+func _set_languages(current_locale: String) -> void:
+	# Technically, Settings.language_names is useless from here on out, but I figure it's probably gonna come in handy in the future
+	var sorted_languages := Array(Settings.language_names)
+	var unsorted_languages := sorted_languages.duplicate()
```

Comment:
What's the purpose of this unsorted array here? Can't you just use the sorted one?

Reply:
The unsorted one is there to get the original index of a language before sorting alphabetically, by using `unsorted_languages.find()`.